src/button-card.ts

```diff
@@ -0,0 +1,558 @@
+          ${nameStateString ? html`<div>${nameStateString}</div>` : ''}
+          `;
+      case 'vertical':
+      default:
+        return html`
+          ${this.config!.show_icon && icon ? html`<ha-icon style="color: ${color}; width: ${this.config!.size}; height: auto;" icon="${icon}" class="${this.rotate(configState)}"></ha-icon>` : ''}
+          ${name ? html`<div>${name}</div>` : ''}
+          ${stateString ? html`<div>${stateString}</div>` : ''}
+          `;
+    }
+  }
+
+  private blankCardColoredHtml(state: HassEntity | undefined): TemplateResult {
+    const color = this.buildCssColorAttribute(state, undefined);
+    const fontColor = this.getFontColorBasedOnBackgroundColor(color);
+    return html`
+      <ha-card class="disabled">
+        <div class="button-card-background-color" style="color: ${fontColor}; background-color: ${color};"></div>
+      </ha-card>
+      `;
+  }
+
+  private cardColoredHtml(state: HassEntity | undefined, configState: StateConfig | undefined): TemplateResult {
+    const color = this.buildCssColorAttribute(state, configState);
+    const fontColor = this.getFontColorBasedOnBackgroundColor(color);
+    const style = this.buildStyle(state, configState);
+    return html`
+      <ha-card class="${this.isClickable(state) ? '' : "disabled"}" @ha-click="${this._handleTap}" @ha-hold="${this._handleHold}" .longpress="${longPress()}" .config="${this.config}">
+        <div class="button-card-background-color" style="color: ${fontColor}; background-color: ${color};">
+          <div class="button-card-main" style="${style}">
+            ${this.buttonContent(state, configState, "inherit")}
+          </div>
+        </div>
+        <mwc-ripple></mwc-ripple>
+      </ha-card>
+      `;
+  }
+
+  private iconColoredHtml(state: HassEntity | undefined, configState: StateConfig | undefined): TemplateResult {
+    const color = this.buildCssColorAttribute(state, configState);
+    const style = this.buildStyle(state, configState);
+    return html`
+      <ha-card class="${this.isClickable(state) ? '' : "disabled"}" @ha-click="${this._handleTap}" @ha-hold="${this._handleHold}" .longpress="${longPress()}" .config="${this.config}">
+        <div class="button-card-main" style="${style}">
+          ${this.buttonContent(state, configState, color)}
+        </div>
+        <mwc-ripple></mwc-ripple>
+      </ha-card>
+      `;
+  }
+
+  public setConfig(config: ButtonCardConfig): void {
+    if (!config) {
+      throw new Error("Invalid configuration");
+    }
+
+    this.config = {
+      tap_action: { action: "toggle" },
+      hold_action: { action: "none" },
+      size: '40%',
+      color_type: 'icon',
+      show_name: true,
+      show_state: false,
+      show_icon: true,
+      show_units: true,
+      ...config
+    };
+    this.config!.default_color = 'var(--primary-text-color)';
```

Comment:
Why put these outside the initial definition?

Reply:
Because that's what I wanted. It's not an option to configure those 😊
Maybe I should define another property for those specifically?

Reply:
No. They can use a theme